### ISSUE TITLE
adding a providedInRoot Service to the providers of a standalone component returns new instance

### ISSUE THREAD

**Issue by LPCmedia** [angular/angular#51508]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

In a standalone component adding a service that is `providedIn: 'root'` to the providers: [] gives us another instance. 
In the case below the service is an initializer. To repro:

### the component

```typescript
@Component({
  standalone: true,
  imports: [AsyncPipe, NgForOf, CommonModule, NgFor],
  providers: [BarService],
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'foo-bar',
  templateUrl: './foo.component.html',
  styleUrls: ['./foo.component.scss'],
})
class FooComponent {
    bars$!: Observable<NotPertinent>;    
    constructor() {
       const myBar: BarService = inject(BarService):
       this.bars$ = myBar.bars$.pipe(
          tap((mocktail) => {
            // nothing !!!
          })
       )
       // subscribe to bars$ here or via async pipe in foo.component.html. 
    }
}
```

### The service 

```typescript
@Injectable({
  providedIn: 'root'
})
export class BarService {

  private httpClient!: HttpClient;
  private barSubject: ReplaySubject<any> = new ReplaySubject(1);
  
  get bars$(): Observable<any> {
    return this.barSubject.asObservable();
  }

  constructor() {
    this.httpClient = inject(HttpClient);
  }

  initBars(): Observable<any> {
    return this.httpClient.get('/api/something/v1/bars')
      .pipe(
        tap((resp) => {
          this.realmsSubject.next(resp);
        })
      )
  }

}

export function barServiceFactory(openBarYesss: BarService) {
  return () => openBarYesss.initBars();
}

export const BAR_INITIALIZERS = [
  {
    provide: APP_INITIALIZER,
    useFactory: barServiceFactory,
    deps: [BarService],
    multi: true,
  },
  BarService,
];

```

### the bootstrapped app config

```typescript
export const appConfig: ApplicationConfig = {
  providers: [
    provideHttpClient(),
    BAR_INITIALIZERS,
  ],

};
```

## Expected 

oneOf : 
- should grab the forRoot instance and thus give me the values so that I can blissfully bind in my template.
OR 
- documentation on how this should work or should NOT work. ( bonus points for clarification on app initializer and providedIn using standalone api's via some kind of guide. )
- perhaps a warning or deprecation notice when I do something like this  if the behavior is intentional. I should get somekind of warning that providers override existing ones when added directly to a component or that the approach is bad. 

## Actual 

- A new instance is included and subscribing to ` bars$` is futile since it never emits from the new instance.
-  I debug, fall into the abyss and change the code from inject() to the classic constructor style and waste time and feel shame.

### Please provide the exception or error you saw

```true
No exception. 
need clarification on if its by design or a bug and a way to find the doc.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 16.2.0
Node: 18.16.0
Package Manager: yarn 1.22.19
OS: darwin arm64

Angular: 16.2.2
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1602.0
@angular-devkit/build-angular   16.2.0
@angular-devkit/core            16.2.0
@angular-devkit/schematics      16.2.0
@angular/cli                    16.2.0
@schematics/angular             16.2.0
rxjs                            7.8.1
typescript                      5.1.6
zone.js                         0.13.1
```

**Comment by pkozlowski-opensource** [comment]
> perhaps a warning or deprecation notice when I do something like this if the behavior is intentional. I should get somekind of warning that providers override existing ones when added directly to a component or that the approach is bad.

Note: this is pre-existing Angular behaviour, not specific to standalone components.

**Comment by LPCmedia** [comment]
> > perhaps a warning or deprecation notice when I do something like this if the behavior is intentional. I should get somekind of warning that providers override existing ones when added directly to a component or that the approach is bad.
> 
> Note: this is pre-existing Angular behaviour, not specific to standalone components.

Yes @pkozlowski-opensource  thank you. I tested that right after and updated the additional info on how I got there :) 
I do think its just a documentation issue. 

> When we use(d) NgModules registering the provider in the component annotation was likely rarely used. And it probably makes sense to have new instance when doing so. With standalone, its perhaps less clear how APP_INITIALIZERS work and since the imports are moved to the component, its natural to do the same with providers. Especially when the app has no routing since the current doc focuses on that.

**Comment by pkozlowski-opensource** [comment]
Chatting about it a bit more in the team - the current behaviour is perfectly expected - the `BarService` is a regular class and it could be place both on the environment provider level as well as on a directive level. The rule here is simple - any class could be provided.

I'm afraid that extending documentation might create more confusion.

**Comment by LPCmedia** [comment]
As mentioned, the current behaviour is fine if this is what is expected. However, Its hard to understand how more explanation / documentation is detrimental to understanding... 

Im sure there is a reason for the behaviour and thats is what I am trying to understand.

**Comment by JoostK** [comment]
There's work underway to extend devtools with insight into DI and providers (#45748) that should help debug such situations.

Regarding documentation, there is [a section to document the pattern of overriding services from within a component](https://angular.io/guide/hierarchical-dependency-injection#scenario-specialized-providers). Our judgement was that additional notes would only appear to complicate and confuse matters, since it may indicate that there are special cases w.r.t. standalone components, but there are none.
